packages/tools/src/utilities/segmentation/getUniqueSegmentIndices.ts

```diff
@@ -0,0 +1,74 @@
+import { Types, cache } from '@cornerstonejs/core';
+import { SegmentationRepresentations } from '../../enums';
+import { getSegmentation } from '../../stateManagement/segmentation/segmentationState';
+import { isVolumeSegmentation } from '../../tools/segmentation/strategies/utils/stackVolumeCheck';
+
+/**
+ * Retrieves the unique segment indices from a given segmentation.
+ *
+ * @param segmentationId - The ID of the segmentation.
+ * @returns An array of unique segment indices.
+ * @throws If no geometryIds are found for the segmentationId.
+ */
+function getUniqueSegmentIndices(segmentationId) {
+  const segmentation = getSegmentation(segmentationId);
+
+  if (segmentation.type === SegmentationRepresentations.Labelmap) {
+    const labelmapData =
+      segmentation.representationData[SegmentationRepresentations.Labelmap];
+
+    const keySet = {};
+
+    if (isVolumeSegmentation(labelmapData)) {
+      const volume = cache.getVolume(segmentationId);
+      const scalarData = volume.getScalarData();
+
+      for (let i = 0; i < scalarData.length; i++) {
+        const segmentIndex = scalarData[i];
+        if (segmentIndex !== 0 && !keySet[segmentIndex]) {
```

Comment:
If you use a set [], then you can just use the add function, then remove just 0 at the end.  Makes this part all 1 line.  If you prefer to leave as an object, then keySet[segmentIndex] ||= true works, and you can still delete the 0 at the end.